### ISSUE TITLE
Rails 5.2.0.rc1 breaks data migrations

### ISSUE THREAD

**Issue by rubiii** [ilyakatz/data-migrate#83]
Just a heads up, not a fix (unfortunately): Migrations have been refactored to better support multiple databases which breaks all data migration tasks.

https://github.com/rails/rails/commit/a2827ec9811b5012e8e366011fd44c8eb53fc714

Result of running the `data:migrate` task:
```
NoMethodError: undefined method 'migrate' for DataMigrate::DataMigrator:Class
.../gems/data_migrate-3.4.0/tasks/databases.rake:249:in 'block (2 levels) in <main>'
```

rails: 5.2.0.rc1
data_migrate: 3.4.0

**Comment by ilyakatz** [comment]
ah, ok, will need to spend some time to figure this out. thanks for the heads up

**Comment by yangtheman** [comment]
I second this, since I've just encountered the same problem.

**Comment by yangtheman** [comment]
@ilyakatz It looks like you would need to use `MigrationContext` instead of `Migrator` => https://github.com/rails/rails/blob/v5.2.0.rc1/activerecord/lib/active_record/migration.rb#L1000  I am not sure simply changing the parent class would make everything work again.

**Comment by ilyakatz** [comment]
thanks for the pointers @yangtheman. i don't think i can change the parent class since it would break for older versions of rails. so i'll probably need to separate them our based on rails version. i haven't had a chance to play with it yet, but will try to find time soon. i see rails 5.2 is making progress fast!

**Comment by ilyakatz** [comment]
@rubiii @yangtheman and whoever is out there, I've added Rails 5.2 support in #84. I haven't released the code into rubygems yet. Can you give it a try in your applications? The branch that the code is on is called `rails5.2`

Thanks in advance!

**Comment by chagar** [comment]
Migrations run successfully for us and all our tests pass.

However, I noticed that some of the other commands don't work properly:

* `bundle exec rake data:rollback` and `bundle exec rake data:migrate:redo results in `NoMethodError: undefined method `rollback' for DataMigrate::DataMigrator:Class`
* `bundle exec rake data:migrate:down VERSION=xxx` migrates the wrong version.
* `bundle exec rake data:migrate:redo VERSION=xxx` reverts multiple migrations, without redoing them, and all the wrong versions.

**Comment by chagar** [comment]
Nice. Migrations up and down and redo work fine now. Thank you. I found some issues with rollback and versions:

* Some of the Error constants need to have "ActiveRecord::" added to them. See PR comments. (uninitialized constant)
* rollback failed when it attempted to rollback the latest version in my data_migrations table, which was a missing file. (Rails db:rollback seems to make it into a no-op when this happens.)
* Later, after running a bunch of other commands, rollback now does nothing, even after deleting the entry in data_migrations that had no file.
* data:version returns the latest version in schema_migrations, i.e. db migrations, not data migrations.

**Comment by ilyakatz** [comment]
@zakmysta @chagar @rubiii @yangtheman @ahey I think I fixed all the outstanding issues (see #84 ). Can you please give `rails5.2` branch another try?

**Comment by Mehonoshin** [comment]
@ilyakatz your fixes work for me

**Comment by chagar** [comment]
It looks like this happens because: (when the latest 'up' migration is different from the latest migration)
* migrator.current_version returns nil in DataMigrate::MigrationContext#move
* because migrator.current_version returns the latest migration even if it is 'down', while MigrationContext#current_version returns the latest 'up' migration as expected.

**Comment by ilyakatz** [comment]
Thanks @chagar, yes, I see what you mean. I noticed that even current version doesn't work that well - I was able to reproduce this `Rails db:rollback seems to make it into a no-op when this happens.` in rails 4.2. I assume you mean `data:rollback, right?

Since this is a current issue and this PR is already pretty heavy, I'd like to separate it into a separate issue (https://github.com/ilyakatz/data-migrate/issues/86). 

There is a lot of refactor that can be done. I tried to improve as I went this rails 5.2 support but that only made me realize there is a lot more to go.
